### PR TITLE
WIP: Initial sample of using resource based authorization

### PR DESCRIPTION
**Pull request by stevejgordon**
Example relating to #982 

NOTE: Do not merge this currently as it's incomplete and just an example at this stage. It would need a little tidy up and tests need to be refactored to support what I've changed. For this example I've just commented out the tests that broke until we can review the concept.

Documentation reference - https://docs.asp.net/en/latest/security/authorization/resourcebased.html

### CODE REVIEW

**Comment by MisterJames** [comment]
@stevejgordon let's chat about this after the standup next week (Aug 20) to figure out where it lands.

**Comment by tonysurma** [comment]
@MisterJames @stevejgordon thoughts on this?  (trying to get PRs cleaned up or out or back on track)  Thanks

**Comment by MisterJames** [comment]
@stevejgordon I know you've done some additional work on this topic on your blog. Any new insight or learnings you wanted to apply before we pick this up again? I think this is the right solution over the extension method, and as pointed out, provides a better test mechanism with lower friction.

**Comment by stevejgordon** [comment]
@MisterJames @tonysurma Have to try and get my head back into this PR but I think it's still the better way to go and follows the pattern laid down in the Microsoft documentation as well.

I've been doing a fair bit of work for an API recently and in that case I've used custom middleware to validate authentication for a specific path route we have. However in that case it's a simple, one profile to one path mapping and easy to check a user's claims match up. Here we have different levels of resource that need checking back org/campaign/event etc.

Overall I think the approach above is probably easier to understand and apply as we grow the application. I'm happy to get this started up again and perhaps get one or two controllers going through this flow as a starting point. If during that it doesn't feel right we can review again. As we've refactored a lot since I put this in I'm thinking I might create a fresh branch and work there.

File: AllReadyApp/Web-App/AllReady/Security/AuthoizationHandlers/OrganizationOperations.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿using Microsoft.AspNet.Authorization.Infrastructure;
+
+namespace AllReady.Security.AuthoizationHandlers
```
Comment (dpaquette) [inline AllReadyApp/Web-App/AllReady/Security/AuthoizationHandlers/OrganizationOperations.cs]: Spelling mistake in namespace here. *AuthorizationHandlers*

**Comment by dpaquette** [comment]
@stevejgordon I am a big fan of this approach in general. I think it really helps to clearly state the resource authorization requirements (e.g. Is this person allowed to edit a campaign?) and makes it much easier to test.

**Comment by stevejgordon** [comment]
I'm going to park this WIP PR since we'll be going through some change for IdentityServer and we can incorporate as necessary during that work

**Pull request closed without merging**